doc/database/multi_namespace_db_instances.md

```diff
@@ -0,0 +1,495 @@
+        self.db_name = dbname
+
+        # Set the separators per namespace as per the database_config file.
+        if(SonicV2Connector.check_instance_valid(self, 'default')):
+            ConfigDBConnector.separators['default'] = self.get_db_separator(self.db_name, 'default')
+        for ns_idx in range(self.connections):
+            if(SonicV2Connector.check_instance_valid(self, str(ns_idx))):
+               ConfigDBConnector.separators[str(ns_idx)] = self.get_db_separator(self.db_name, str(ns_idx))
+            else:
+                ConfigDBConnector.separators[str(ns_idx)] = '|'
+
+        SonicV2Connector.connect(self, self.db_name, retry_on, namespace)
+        if wait_for_init:
+            self.__wait_for_db_init()
+
+    """
+    Connect to the DB instance specified by the namespace parameter in database_config.json file.
+	* "default", it connects to the default instances of "db_name".
+	* <0...n>, any value between 0 and n, it connects to "db_name" instance in that namespace as per the INCLUDES
+	* "all", it connects to all "db_name" databases both default and in all namespaces added in INCLUDES.
+    """
+    def connect(self, wait_for_init=True, retry_on=False, namespace='default'):
+        self.cfg_db_connect('CONFIG_DB', wait_for_init, retry_on, namespace)
+
+    def get_entry(self, table, key, namespace='default'):
+        """Read a table entry from config db.
+        Args:
+            table: Table name.
+            key: Key of table entry, or a tuple of keys if it is a multi-key table.
+        Returns:
+            Table row data in a form of dictionary {'column_key': 'value', ...}
+            Empty dictionary if table does not exist or entry does not exist.
+        """
+        got_data = {}
+        instances = []
+        if namespace == 'all':
+            instances = self.get_redis_client_instances()
+        else:
+            instances.append(namespace)
+
+        for instance in instances:
+            key = self.serialize_key(key, instance)
+            _hash = '{}{}{}'.format(table.upper(), ConfigDBConnector.separators[instance], key)
+            client = self.get_redis_client(self.db_name, instance)
+            if client == None:
+                continue;
+
+            data = self.__raw_to_typed(client.hgetall(_hash))
+            if data:
+               got_data.update(data)
+
+        return got_data
+
+
+```
+Note: There are many other utilities which needs this additional parameter, will be updated in a similar fashion.
+
+## Design of C++ Interface :  DBConnector()
+The C++ DBConnector interface is mainly used in the process context of swss/syncd etc in the dockers in their respective namespaces. In the namespaces currently we don't have any external DB refereces, hence not planning to extend the C++ DBConnector to handle multiple database_config.json files.
+
+Will be taken up as the next phase activity to get this class also at par with the python SonicDBConfig/SonicV2Connector modules.
```

Comment:
this is a must because we are deprecating the pure swsssdk library and plan to python-swsscommon library for all python related applications.

Reply:
Updated the doc, will take up the changes to DBConnector class as well and raise the PR for those changes.